tests/trainer/flags/test_env_vars.py

```diff
@@ -53,4 +53,4 @@ def test_passing_env_variables_devices(cuda_available_mock, device_count_mock):
     trainer = Trainer()
     assert trainer.devices == 2
     trainer = Trainer(accelerator="gpu", devices=1)
-    assert trainer.devices == 1
+    assert trainer.devices == 2
```

Comment:
what's happening here? can you add a comment?

Reply:
This is a bug fix, the L48 above has "@mock.patch.dict(os.environ, {"CUDA_VISIBLE_DEVICES": "0,1", "PL_TRAINER_GPUS": "2"})" which equal to set `gpus=2` in trainer, which has higher priority compare to `device=1`, so the `device=2` here

Reply:
But that's inconsistent with the other tests above in this file

this was added in https://github.com/PyTorchLightning/pytorch-lightning/pull/6510

cc @Borda

Reply:
It is consistent. Here : https://github.com/PyTorchLightning/pytorch-lightning/blob/25b505508d376d87103a592f23f7c9b0a602d12e/tests/accelerators/test_accelerator_connector.py#L542-L550

Reply:
Ah okay, I know why this is happening. The problem is that #11697 changed `gpus=1` to `devices=1` but did not change `PL_TRAINER_GPUS` to `PL_TRAINER_DEVICES`

The test should be:

```python
-@mock.patch.dict(os.environ, {"CUDA_VISIBLE_DEVICES": "0,1", "PL_TRAINER_GPUS": "2"})
+@mock.patch.dict(os.environ, {"CUDA_VISIBLE_DEVICES": "0,1", "PL_TRAINER_DEVICES": "2"})
@mock.patch("torch.cuda.device_count", return_value=2)
@mock.patch("torch.cuda.is_available", return_value=True)
def test_passing_env_variables_devices(cuda_available_mock, device_count_mock):
    """Testing overwriting trainer arguments."""
    trainer = Trainer()
    assert trainer.devices == 2
    trainer = Trainer(accelerator="gpu", devices=1)
-    assert trainer.devices == 2
+    assert trainer.devices == 1
```

Reply:
This is not direct override, like some of the tests above.
This one is env var is `gpus=2`, but overrides is `device=1`, so trainer get both `gpus=2, device=1`
I tested local, if here we do `trainer = Trainer(accelerator="gpu", gpus=3)`, we will get `assert trainer.devices == 3`

Reply:
oh! The bug introduced here: https://github.com/PyTorchLightning/pytorch-lightning/commit/b34d8673a9e77d25a826b7ada31d1204dfc818ab
Don't know why ci didn't catch it

Reply:
@carmocca found the same thing :)